src/app/ipv/middleware.js

```diff
@@ -227,15 +227,30 @@ function handleAppStoreRedirect(req, res, next) {
   }
 }
 
+async function handleUnexpectedPage(req, res, pageId) {
+  logError(
+    req,
+    {
+      pageId: pageId,
+      expectedPage: req.session.currentPage,
+    },
+    "page :pageId doesn't match expected session page :expectedPage",
+  );
+
+  req.session.currentPage = "pyi-attempt-recovery";
+  return await saveSessionAndRedirect(
+    req,
+    res,
+    `/ipv/page/pyi-attempt-recovery`,
+  );
+}
+
 module.exports = {
   renderAttemptRecoveryPage: async (req, res) => {
     res.render("ipv/page/pyi-attempt-recovery.njk", {
       csrfToken: req.csrfToken(),
     });
   },
-  // This method is currently only used by a link on the pyi-f2f-delete-details page.
-  // It shouldn't be used for anything else, see
-  // https://govukverify.atlassian.net/browse/PYIC-4859.
   updateJourneyState: async (req, res, next) => {
```

Comment:
I removed this comment as it is no longer true, we are now using this as a handler for anchor tag driven journey events, and relying on backend validation to ensure that it isn't abused.